src/tests/e2etest/Tests.cs

```diff
@@ -0,0 +1,126 @@
+
+        [Test]
+        public void CommandRunnerTest_ActionRunCommand_echo_HelloWorld()
+        {
+            var random = Convert.ToBase64String(Guid.NewGuid().ToByteArray()).Substring(0, 4);
+            var desiredCommand = new CommandArguments
+            {
+                CommandId = random,
+                Arguments = "echo HelloWorld",
+                Action = Action.ActionRunCommand,
+                Timeout = 60,
+                SingleLineTextResult = true
+            };
+            var expectedCommandStatus = new CommandStatus
+            {
+                CommandId = random,
+                ResultCode = 0,
+                ExtendedResultCode = 0,
+                TextResult = "HelloWorld ",
+                CurrentState = CommandState.Succeeded,
+            };
+
+            var twinPatch = CreateCommandArgumentsPropertyPatch("CommandRunner", desiredCommand);
+
+            if (UpdateTwinBlockUntilUpdate(twinPatch))
+            {
+                var deserializedTwin = JsonSerializer.Deserialize<CommandStatus>(GetTwin().Properties.Reported["CommandRunner"]["CommandStatus"].ToString());
+                // Wait until our random commandId is present in reported properties
+                DateTime startTime = DateTime.Now;
+                while(deserializedTwin.CommandId != random && (DateTime.Now - startTime).TotalSeconds < 30)
+                {
+                    Console.WriteLine("[CommandRunnerTests] waiting for commandId to be equivalent...");
+                    Task.Delay(twinRefreshIntervalMs).Wait();
+                    deserializedTwin = JsonSerializer.Deserialize<CommandStatus>(GetNewTwin().Properties.Reported["CommandRunner"]["CommandStatus"].ToString());
+                }
+
+                AreEqualByJson(expectedCommandStatus, deserializedTwin);
+            }
+            else
+            {
+                Assert.Fail("Timeout for updating twin - ActionRunCommand");
+            }
+        }
+    }
+
+    public class FirewallTests : E2eTest
+    {
+        string ComponentName = "Firewall";
+        public enum FirewallStateCode
+        {
+            Unknown = 0,
+            Enabled,
+            Disabled,
+        }
+
+        public partial class Firewall
+        {
+        public FirewallStateCode FirewallState { get; set; }
+        public string FirewallFingerprint { get; set; }
+        }
+        public partial class ExpectedFirewallPattern
+        {
+            public string FirewallState { get; set; }
+            public string FirewallFingerprint { get; set; }
+        }
+
+        [Test]
+        public void FirewallTest_Regex()
+        {
+            var expectedFirewallPattern = new ExpectedFirewallPattern
+            {
+                FirewallState = "[0-2]",
+                FirewallFingerprint = "[0-9a-z]{64}"
+            };
+
+            var deserializedReportedObject = JsonSerializer.Deserialize<Firewall>(GetTwin().Properties.Reported[ComponentName].ToString());
```

Comment:
Rename GetTwin to GetObject or similar

Reply:
Since the base APIs use the name "twin", I will also keep this as is.